admin-api/oauth.md

```diff
@@ -0,0 +1,91 @@
+| **Access Token**         | A token used to access protected resources                                                                                               |
+| **Resource Server**      | Server hosting the protected resources capable of accepting and responding to requests using access tokens.                              |
+| **Authorization Server** | A server which issues access tokens after successfully authenticating a client and resource owner, and authorizing the request.          |
+| **Scope**                | A permission that grants access to one or several resources.                                                                             |
+| **JWT Token**            | Or JSON Web Token  is a method for representing claims securely between two parties.                                                     |
+| **Grant Type**           | The protocol has several types or workflow to validate authentication known as grant types (Authorization Code, Client Credentials, ...) |
+
+## Our use case
+
+In the use case of the Admin API, here is how each element can be identified:
+
+- **Resource owner**: The merchant/administrator of the PrestaShop website.
+- **Client**: An OMS or a PIM tool that needs to integrate with the PrestaShop website.
+- **Resource server**: The PrestaShop website.
+- **Authorization Server**: The PrestaShop website.
+
+### Grant Type: Client Credentials
+
+For "server-to-server" interactions, the recommended OAuth workflow is the client credentials workflow. The client is provided with a client ID and a client secret, the context of execution of the API is not as a user.
+
+The operations made via this workflow will be associated to an application not a user.
+
+The client is considered trusty so the workflow is simple and require very few steps.
+
+{{% notice note %}}
+In the default use case the website is acting both as the Resource server AND the Authorization server, but you can also use other solutions for your authorization server (see External Authorization Server), however the website will always be the Resource Server.
+{{% /notice %}}
+
+## Client Credential Workflow
+
+The **client credential** workflow is quite simple, the client only needs to perform one request to the Authorization server, it needs to specify:
+
+- **Client ID**
+- **Client secret**
+- **Grant type**: `client_credentials`
+- **Scopes**: the list of scopes that will be usable via the received access token
+
+{{% notice note %}}
+Keep in mind that the first request to get access token already has to define the scopes (or permissions) that will be granted with the request access token, if you need to perform different API calls with different scopes you must provide the whole required scopes for all the subsequent calls.
+
+It is recommenced to use a limited subset of scopes in case your access token is stolen so that it doesn't grant too many permissions.
```

Comment:
```suggestion
Keep in mind that the first request to get an access token already has to define the scopes (or permissions) that will be granted with the request access token. If you need to perform different API calls with different scopes, you must provide all the required scopes for all the subsequent calls.

It is recommended to use a limited subset of scopes in case your access token is stolen so that it only grants a few permissions.
```